test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -2251,3 +2251,131 @@
+                  (is (nil? (.verify cert capub)))
+                  (testing "not after date should be ~90 days in the future"
+                    (let [diff (- (.getTime (.getNotAfter cert)) (.getTime (Date.)))
+                          days (.convert TimeUnit/DAYS diff TimeUnit/MILLISECONDS)]
+                      ;; 89 days plus some number of hour, minutes seconds
+                      (is (= 89 days))))))
+              ;; certificate should be in inventory
+              (testing "should be in the inventory"
+                (let [inventory (slurp inventory-file)
+                      entries (string/split inventory #"\n")]
+                  (is (= (count entries) 1))
+                  (let [row (string/split (first entries) #" ")]
+                    ;; row is constructed of serial-number, start date, end date and name
+                    ;; sandbox has some used serial numbers
+                    (is (= "0x0003" (first row)))
+                    (is (= (str "/" (:subject csr-info)) (nth row 3))))))))))
+      (testing "correctly rejects a non-existent csr"
+        (let [random-csr-name (ks/rand-str :alpha-digits 8)]
+          (is (= {:signed     []
+                  :not-signed [random-csr-name]}
+                 (ca/sign-multiple-certificate-signing-requests! [random-csr-name] settings report-activity)))))
+      (testing "correctly rejects a cert with Subject alternative names"
+        (let [alt-name-ext {:oid      utils/subject-alt-name-oid
+                            :value    {:dns-name ["bad-name"]}
+                            :critical false}
+              csr-info (generate-csr settings [alt-name-ext] [{:oid ca/pp_auth_auto_renew-attribute :value true}])]
+          (is (= {:signed     []
+                  :not-signed [(:subject-name csr-info)]}
+                 (ca/sign-multiple-certificate-signing-requests! [(:subject-name csr-info)] settings report-activity)))))
+      (testing "correctly rejects a cert with authorization extensions when disabled"
+        (let [csr-info (generate-csr settings [{:oid ca/ppAuthCertExt :value "true" :critical false}] [{:oid ca/pp_auth_auto_renew-attribute :value true}])]
+          (is (= {:signed     []
+                  :not-signed [(:subject-name csr-info)]}
+                 (ca/sign-multiple-certificate-signing-requests! [(:subject-name csr-info)] settings report-activity)))))
+      (testing "correctly rejects a cert with unapproved extensions"
+        (let [csr-info (generate-csr settings [{:oid "1.9.9.9.9.9.0" :value "true" :critical false}] [{:oid ca/pp_auth_auto_renew-attribute :value true}])]
+          (is (= {:signed     []
+                  :not-signed [(:subject-name csr-info)]}
+                 (ca/sign-multiple-certificate-signing-requests! [(:subject-name csr-info)] settings report-activity))))))
+    (testing "multiple entry with both bad and good csrs"
+      (let [count-range (range 0 100)
+            _ (println "begin generate good csrs")
```

Comment:
This takes a little while (a few seconds), so I left the print statements in here to make it easier to understand why it is delayed.